src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -128,12 +131,81 @@ public DefinitionRegistrationOptions GetRegistrationOptions(DefinitionCapability
                 new LocationOrLocationLink(new Location
                 {
                     Uri = originComponentUri,
-                    Range = new Range(new Position(0, 0), new Position(0, 0)),
+                    Range = range,
                 }),
             });
         }
 
-        internal static async Task<TagHelperBinding?> GetOriginTagHelperBindingAsync(
+        private async Task<Range> GetNavigateRangeAsync(DocumentSnapshot documentSnapshot, BoundAttributeDescriptor? attributeDescriptor, CancellationToken cancellationToken)
+        {
+            if (attributeDescriptor is not null)
+            {
+                var originCodeDocument = await documentSnapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
+                var range = await TryGetPropertyRangeAsync(originCodeDocument, attributeDescriptor.GetPropertyName(), _documentMappingService, cancellationToken).ConfigureAwait(false);
+
+                if (range is not null)
+                {
+                    return range;
+                }
+            }
+
+            return new Range(new Position(0, 0), new Position(0, 0));
+        }
+
+        internal static async Task<Range?> TryGetPropertyRangeAsync(RazorCodeDocument codeDocument, string propertyName, RazorDocumentMappingService documentMappingService, CancellationToken cancellationToken)
```

Comment:
Nit: Generally when something is named `TryXYZ` I would guess that it returns bool (or Task<bool>) and has an out  parameter. This has the benefit of letting you automate the nullable-check (as long as you apply attributes to the out parameter.)

Reply:
Can't have out parameters on Async methods though ☹️